compiler/hash-untyped-semantics/src/visitor.rs

```diff
@@ -218,7 +218,11 @@ impl<'s> SemanticAnalyser<'s> {
             // Check that the supplied argument to a function modifying directive
             // is a declaration of a function that the directive will apply to.
             self.validate_data_def_directive(directive, subject)
-        } else if !directive.is(IDENTS.dump_ast) && !directive.is(IDENTS.dump_tir) {
+        } else if !directive.is(IDENTS.dump_ast)
+            && !directive.is(IDENTS.dump_tir)
+            && !directive.is(IDENTS.run)
+            && !directive.is(IDENTS.pure)
```

Comment:
Surely this should check that is set on a function definition, or is this checked later?

Reply:
Fixed